lib/child_process.js

```diff
@@ -136,7 +136,7 @@ function fork(modulePath, args = [], options) {
   if (options != null) {
     validateObject(options, 'options');
   }
-  options = { ...options, shell: false };
+  options = { __proto__: null, ...options, shell: false };
```

Comment:
Not related to this PR, but I wonder if we should use `{ __proto__: options, shell: false }` and let users decide if they want prototype pollution or not.

Reply:
Wouldn't it make this object affected by subsequent mutations outside?
In fact, I wonder if we should use `structuredClone()` or at least make copies of object properties like `.env` to avoid race conditions if multiple `fork()`s are called with reused options.

Reply:
> Wouldn't it make this object affected by subsequent mutations outside?

I guess it depends if we read the properties in the same tick or not (which I expect we do most of the time, but I haven't checked).


> In fact, I wonder if we should use `structuredClone()`

That wouldn't help with inconsistency of our API: sometimes, we dismiss inherited properties, sometimes we don't. But it's clearly not a big deal as I don't think we ever received any bug report regarding that.

> or at least make copies of object properties like `.env` to avoid race conditions if multiple `fork()`s are called with reused options.

but wouldn't folks expect the same env to be shared if they supply the same object? The fact that no one complains makes me think we're overthinking it 😅